그림퀴즈 PR

dalbodeule opened the pull request:
이전에 모리끄투 등 일부 서버에서 사용중인 그림퀴즈를 공개합니다.

끄투코리아의 그림퀴즈 소스와는 완전 다른 소스입니다.

dalbodeule committed:
그림퀴즈 업데이트

dalbodeule committed:
그림퀴즈 점수 반영

dalbodeule committed:
미처 올리지 못한 이미지 업로드

small-sunshines commented:
그림퀴즈 소스에서 파일 경로 관련 오류발견

dalbodeule committed:
경로 수정

dalbodeule committed:
CSS가 누락되었던것같습니다.

small-sunshines commented:
실수로 PC버전의 CSS가 누락되었던것같습니다.

dalbodeule committed:
in_kkutu.css 가 올바른 파일, in_game_kkutu.css 원상복구 및 background-color 수정

dalbodeule committed:
관전자에게 Painter가 주어지던 오류 수정

lshqqytiger commented:
시간이 초과되면 오류를 뿜으며 서버가 재부팅됩니다.

[2019-01-09 22:29:59] ERROR OCCURRED ON THE MASTER! :8496 [2019-1-9 22:29:59] ERROR: TypeError: Cannot read property 'NaN' of undefined TypeError: Cannot read property 'NaN' of undefined at exports.Room.turnHint (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:113:21) at Timeout._onTimeout (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:100:55) at ontimeout (timers.js:436:11) at tryOnTimeout (timers.js:300:5) at listOnTimeout (timers.js:263:5) at Timer.processTimers (timers.js:223:10)
[2019-01-09 22:30:00] ERROR OCCURRED! This worker will die in 10 seconds. :8496 [2019-1-9 22:29:59] ERROR: TypeError: Cannot read property 'NaN' of undefined TypeError: Cannot read property 'NaN' of undefined at exports.Room.turnHint (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:113:21) at Timeout._onTimeout (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:100:55) at ontimeout (timers.js:436:11) at tryOnTimeout (timers.js:300:5) at listOnTimeout (timers.js:263:5) at Timer.processTimers (timers.js:223:10)
[2019-01-09 22:30:00] New user #guest__YMReghPgN1C1Qaba183XTTlYSwhbV8Yc
[2019-01-09 22:30:03] ERROR OCCURRED ON THE MASTER! :8496 [2019-1-9 22:30:03] ERROR: TypeError: Cannot read property 'NaN' of undefined TypeError: Cannot read property 'NaN' of undefined at exports.Room.turnHint (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:113:21) at Timeout._onTimeout (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:101:55) at ontimeout (timers.js:436:11) at tryOnTimeout (timers.js:300:5) at listOnTimeout (timers.js:263:5) at Timer.processTimers (timers.js:223:10)
[2019-01-09 22:30:03] ERROR OCCURRED! This worker will die in 10 seconds. :8496 [2019-1-9 22:30:03] ERROR: TypeError: Cannot read property 'NaN' of undefined TypeError: Cannot read property 'NaN' of undefined at exports.Room.turnHint (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:113:21) at Timeout._onTimeout (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:101:55) at ontimeout (timers.js:436:11) at tryOnTimeout (timers.js:300:5) at listOnTimeout (timers.js:263:5) at Timer.processTimers (timers.js:223:10)

junykkutu commented:
이 소스를 사용해도 되나요?

lshqqytiger commented:
버그가 있을 수 있습니다.

kdhkr committed:
그림퀴즈 소스 내 일부 사항 수정

1. 플레이어가 연습 모드로 플레이 도중 중도 퇴장 시 힌트 타이머가 멈추지 않아 발생한 오류 수정
2. "짧음" 특수 규칙 추가
3. 힌트 타이머 간격이 비정상적으로 짧았던 점 수정

kdhkr committed:
특수 규칙 "짧음" 추가 (ko_KR.json)

그림퀴즈에서 나오는 단어의 길이를 최대 4글자로 조절하는 규칙입니다.

kdhkr committed:
특수 규칙 "짧음" 추가 (en_US.json)

그림퀴즈에서 나오는 단어의 길이를 최대 4글자로 조절하는 규칙입니다.

kdhkr committed:
특수 규칙 "짧음" 추가 (const.js)

그림퀴즈에서 나오는 단어의 길이를 최대 4글자로 조절하는 규칙입니다.

hatty163 commented:
> 시간이 초과되면 오류를 뿜으며 서버가 재부팅됩니다.
> 
> [2019-01-09 22:29:59] ERROR OCCURRED ON THE MASTER! :8496 [2019-1-9 22:29:59] ERROR: TypeError: Cannot read property 'NaN' of undefined TypeError: Cannot read property 'NaN' of undefined at exports.Room.turnHint (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:113:21) at Timeout._onTimeout (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js💯55) at ontimeout (timers.js:436:11) at tryOnTimeout (timers.js:300:5) at listOnTimeout (timers.js:263:5) at Timer.processTimers (timers.js:223:10)
> [2019-01-09 22:30:00] ERROR OCCURRED! This worker will die in 10 seconds. :8496 [2019-1-9 22:29:59] ERROR: TypeError: Cannot read property 'NaN' of undefined TypeError: Cannot read property 'NaN' of undefined at exports.Room.turnHint (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:113:21) at Timeout._onTimeout (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js💯55) at ontimeout (timers.js:436:11) at tryOnTimeout (timers.js:300:5) at listOnTimeout (timers.js:263:5) at Timer.processTimers (timers.js:223:10)
> [2019-01-09 22:30:03] ERROR OCCURRED ON THE MASTER! :8496 [2019-1-9 22:30:03] ERROR: TypeError: Cannot read property 'NaN' of undefined TypeError: Cannot read property 'NaN' of undefined at exports.Room.turnHint (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:113:21) at Timeout._onTimeout (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:101:55) at ontimeout (timers.js:436:11) at tryOnTimeout (timers.js:300:5) at listOnTimeout (timers.js:263:5) at Timer.processTimers (timers.js:223:10)
> [2019-01-09 22:30:03] ERROR OCCURRED! This worker will die in 10 seconds. :8496 [2019-1-9 22:30:03] ERROR: TypeError: Cannot read property 'NaN' of undefined TypeError: Cannot read property 'NaN' of undefined at exports.Room.turnHint (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:113:21) at Timeout._onTimeout (C:\Users\user\Desktop\Private\KKuTu-master\Server\lib\Game\games\drawing.js:101:55) at ontimeout (timers.js:436:11) at tryOnTimeout (timers.js:300:5) at listOnTimeout (timers.js:263:5) at Timer.processTimers (timers.js:223:10)

@lshqqytiger 해당 오류를 인지하였고, 수정 절차에 들어갔습니다. @dalbodeule님 레프지토리 해당 Branch에 이와 기타 잡것을 수정하는 Pull Request를 넣어 놓았으니, 참고하시면 좋겠네요 :D

dalbodeule committed:
Merge pull request #3 from hatty163/patch-5

그림퀴즈 소스 내 일부 사항 수정

pprcYT commented:
![image](https://user-images.githubusercontent.com/63226197/91410682-3b822980-e882-11ea-8a76-2a21a3db5d92.png)
이런식으로 옆으로 치우칩니다.(저만 그런가요?)

lshqqytiger commented:
![1](https://user-images.githubusercontent.com/39524005/91414898-2b6d4880-e888-11ea-9fc9-77d68ab54525.PNG)
저는 정상이네요.

dalbodeule committed:
DrawQuiz image data diff

- add image data diff on DrawQuiz
- server and client will send/recive diff data
- if diff data not valid, server and client will request/recive full data
- and fix log system (if data not json, crash server)

dalbodeule committed:
Merge remote-tracking branch 'dalbodeule/DrawQuiz' into DrawQuiz

dalbodeule commented:
현재 작동 여부가 검증되지 않은 최적화 패치가 Push 되었습니다.

중요: **일부 파일이 Conflict를 일으키고 있습니다.**

dalbodeule committed:
DrawQuiz optimize patch bug solve

dalbodeule commented:
이번 그림퀴즈 최적화 패치는 diff를 이용하여 그림 데이터 중 바뀐 부분만 서버와 클라이언트 간 전달하도록 만들었습니다.

그림 데이터 중 바뀐 부분만 전달하도록 하여 데이터 통신량을 줄일 수 있도록 하였습니다.

데이터의 유실에 대비하여 서버나 플레이어가 요청할 경우 원래 데이터를 요청/전달할 수 있게 되어있습니다.

githubmahan commented:
머지를 해주세요

ghost commented:
제가 수정을 잘못해서 일수도 있겠지만
한가지 색상으로 그림을 그렸을때 다른 색상을 사용하기 전까지는 다른사람에게 표시되지 않는것 같습니다.

https://github.com/JJoriping/KKuTu/assets/62054933/0bd84d74-9f9c-4b3e-b89c-9590aa7010b4

jhj1228 commented:
> 제가 수정을 잘못해서 일수도 있겠지만 한가지 색상으로 그림을 그렸을때 다른 색상을 사용하기 전까지는 다른사람에게 표시되지 않는것 같습니다.

문제가 없는 것 같습니다.

ghost commented:
혹시 다른 색도 써보셨나요? 검은색은 잘되더라고요.

jhj1228 commented:
다른색으로 확인 결과 처음에는 인식이 되지 않다가 다른 색깔로 변경 후 다시 변경하면 정상적으로 표시되는 것 같습니다.

kkutuwings commented:
![스크린샷 2023-12-06 001450](https://github.com/JJoriping/KKuTu/assets/151385288/a3ed04ac-d24d-4f70-ae9c-4f663f3e8859)
전부다 적용하였지만 콘솔에 해당 오류가 뜹니다

이것만이 아니라 게임중 몇초인지 안뜨고,초가 지나면 문제 발생으로 튕깁니다.

lshqqytiger commented:
[fabric.min.js](https://github.com/JJoriping/KKuTu/pull/326/files#diff-ceebc77e0f44826312a5a6ddedbccd159a525fa0b3d2dd24a21e4e16c072f28a) 파일이 누락된 것으로 보입니다.

kkutuwings commented:
![스크린샷 2023-12-06 154741](https://github.com/JJoriping/KKuTu/assets/151385288/bcaa51ca-6838-410b-bb83-2a8de75ff362)

fabric.min.js를 직접 다운로드 시켜서 했지만 이런 오류가 뜹니다.

lshqqytiger commented:
[이 줄](https://github.com/JJoriping/KKuTu/pull/326/files#diff-f237728a40a4abed0627e6a99bfa6d976ec398edc8fcf485f470530aadb1eb25R45)을 추가하셨나요?

kkutuwings commented:
![스크린샷 2023-12-06 224203](https://github.com/JJoriping/KKuTu/assets/151385288/89e038ce-12bf-42ad-a32e-b0a75f15c81c)
일단 안 그려지는 것은 해결하였습니다.
그런데 그리면 상대에게 그린 내용이 전달이 안됩니다.
콘솔 내용 첨부하겠습니다.

lshqqytiger commented:
같은 유형의 문제입니다. [kkutu.pug](https://github.com/JJoriping/KKuTu/pull/326/files#diff-f237728a40a4abed0627e6a99bfa6d976ec398edc8fcf485f470530aadb1eb25R46) [differ](https://github.com/JJoriping/KKuTu/pull/326/files#diff-fb851ea24e3800a71f8e61470902898e6e74a0bdc659edf9e9e9759cdfd95494)

kkutuwings commented:
kkutu.pug코드는 저번에 수정할때 같이 추가했습니다. 그런데 저런 오류가 납니다.

lshqqytiger commented:
[diff-match-patch-compressed.js](https://github.com/JJoriping/KKuTu/pull/326/files#diff-fb851ea24e3800a71f8e61470902898e6e74a0bdc659edf9e9e9759cdfd95494) 파일이 제대로 로드되고 있는지 Network 탭에서 확인해보세요.

pasunyan commented:
오류가 뜨면서 게임 서버 실행이 안됩니다
터미널 내용 첨부합니다

 internal/modules/cjs/loader.js:818
  throw err;
  ^

Error: Cannot find module 'diff-match-patch'
Require stack:
- /root/KKuTu/Server/lib/Game/kkutu.js
- /root/KKuTu/Server/lib/Game/master.js
- /root/KKuTu/Server/lib/Game/slave.js
- /root/KKuTu/Server/lib/Game/cluster.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/root/KKuTu/Server/lib/Game/kkutu.js:24:24)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/root/KKuTu/Server/lib/Game/kkutu.js',
    '/root/KKuTu/Server/lib/Game/master.js',
    '/root/KKuTu/Server/lib/Game/slave.js',
    '/root/KKuTu/Server/lib/Game/cluster.js'
  ]
}

lshqqytiger commented:
package.json에 변경 사항이 있는 PR의 경우 `npm install`을 해주셔야 합니다.

pasunyan commented:
> 오류가 뜨면서 게임 서버 실행이 안됩니다 터미널 내용 첨부합니다
> 
> internal/modules/cjs/loader.js:818 throw err; ^
> 
> Error: Cannot find module 'diff-match-patch' Require stack:
> 
> * /root/KKuTu/Server/lib/Game/kkutu.js
> * /root/KKuTu/Server/lib/Game/master.js
> * /root/KKuTu/Server/lib/Game/slave.js
> * /root/KKuTu/Server/lib/Game/cluster.js
>   at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
>   at Function.Module._load (internal/modules/cjs/loader.js:667:27)
>   at Module.require (internal/modules/cjs/loader.js:887:19)
>   at require (internal/modules/cjs/helpers.js:74:18)
>   at Object. (/root/KKuTu/Server/lib/Game/kkutu.js:24:24)
>   at Module._compile (internal/modules/cjs/loader.js:999:30)
>   at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
>   at Module.load (internal/modules/cjs/loader.js:863:32)
>   at Function.Module._load (internal/modules/cjs/loader.js:708:14)
>   at Module.require (internal/modules/cjs/loader.js:887:19) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [
>   '/root/KKuTu/Server/lib/Game/kkutu.js',
>   '/root/KKuTu/Server/lib/Game/master.js',
>   '/root/KKuTu/Server/lib/Game/slave.js',
>   '/root/KKuTu/Server/lib/Game/cluster.js'
>   ]
>   }
모듈 미설치 문제였습니다 
해결하였습니다

aslhanball commented:
이런 오류가 뜹니다

[2024-01-12 21:30:45] ERROR OCCURRED! This worker will die in 10 seconds. :8496 [2024-1-12 9:30:45 ├F10: PM┤] ERROR: Error: Cannot find module '../const' Require stack: - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\games\drawing.js - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\kkutu.js - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\master.js - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\slave.js - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\cluster.js Error: Cannot find module '../const' Require stack: - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\games\drawing.js - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\kkutu.js - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\master.js - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\slave.js - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\cluster.js at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15) at Function.Module._load (internal/modules/cjs/loader.js:667:27) at Module.require (internal/modules/cjs/loader.js:887:19) at require (internal/modules/cjs/helpers.js:74:18) at Object.<anonymous> (C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\games\drawing.js:25:15) at Module._compile (internal/modules/cjs/loader.js:999:30) at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10) at Module.load (internal/modules/cjs/loader.js:863:32) at Function.Module._load (internal/modules/cjs/loader.js:708:14) at Module.require (internal/modules/cjs/loader.js:887:19)

lshqqytiger commented:
`Server\lib\Game\games\drawing.js` 파일의 19번째 줄 부근의 내용이 어떻게 되나요?

aslhanball commented:
> `Server\lib\Game\games\drawing.js` 파일의 19번째 줄 부근의 내용이 어떻게 되나요?

/** 라고 써있습니다

lshqqytiger commented:
저는 **부근**이라고 말씀드렸습니다. 그리고 19번째 줄에 `/**`이 있으면 제대로 병합하신 것이 아닙니다.

aslhanball commented:
* You should have received a copy of the GNU General Public License
 * along with this program. If not, see .
 */

/**
 * 볕뉘 수정사항:
 * var 에서 let/const 로 변수 변경
 * kkutu-lib 모듈에 호환되도록 수정
 */

const Const = require('../const');
const lib = require('kkutu-lib');
const Lizard = lib.lizard;
let DB;
let DIC;

exports.init = function(_DB, _DIC){
	DB = _DB;
	DIC = _DIC;
};
exports.getTitle = function(){
	const R = new Lizard.Tail();
	const my = this;
	
	my.game.done = [];
	setTimeout(function(){
		R.go("①②③④⑤⑥⑦⑧⑨⑩");
	}, 500);
	return R;

lshqqytiger commented:
최근 commit까지 전부 병합하세요. [참고](https://github.com/lshqqytiger/KKuTuManual/blob/main/PR%20%EB%B3%91%ED%95%A9.md)

aslhanball commented:
internal/modules/cjs/loader.js:818 throw err; ^ Error: Cannot find module 'diff-match-patch' Require stack: - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\kkutu.js - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\master.js - C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\cluster.js at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15) at Function.Module._load (internal/modules/cjs/loader.js:667:27) at Module.require (internal/modules/cjs/loader.js:887:19) at require (internal/modules/cjs/helpers.js:74:18) at Object.&lt;anonymous> (C:\Users\모자이크\Desktop\KKuTu-master\KKuTu-master\Server\lib\Game\kkutu.js:24:24) at Module._compile (internal/modules/cjs/loader.js:999:30) at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10) at Module.load (internal/modules/cjs/loader.js:863:32) at Function.Module._load (internal/modules/cjs/loader.js:708:14) at Module.require (internal/modules/cjs/loader.js:887:19) { code: 'MODULE_NOT_FOUND', requireStack: [ 'C:\\Users\\모자이크\\Desktop\\KKuTu-master\\KKuTu-master\\Server\\lib\\Game\\kkutu.js', 'C:\\Users\\모자이크\\Desktop\\KKuTu-master\\KKuTu-master\\Server\\lib\\Game\\master.js', 'C:\\Users\\모자이크\\Desktop\\KKuTu-master\\KKuTu-master\\Server\\lib\\Game\\cluster.js' ] }
G: CLOSED WITH CODE 1
[2024-01-12 22:50:03] &lt;&lt; KKuTu Web >>
[2024-01-12 22:50:04] OAuth Strategy naver loaded successfully.
	[2024-01-12 22:50:04] OAuth Strategy discord loaded successfully.
	
	라고 뜹니다...

lshqqytiger commented:
[바로 위](https://github.com/JJoriping/KKuTu/pull/326#issuecomment-1873796260)를 보세요.

aslhanball commented:
그 죄송하지만 cmd에 npm install를 쓰는건가요?

aslhanball commented:
그 너무 질문을 많이하는것같아 죄송합니다..

npm 설치 후  npm의 에러는 수정되었지만 

[2024-01-13 11:23:16] ERROR OCCURRED! This worker will die in 10 seconds. :8496 [2024-1-13 11:23:16 ├F10: AM┤] ERROR: Error: Unknown authenticationOk message typeMessage { name: 'authenticationOk', length: 23 } Error: Unknown authenticationOk message typeMessage { name: 'authenticationOk', length: 23 } at Connection.parseR (C:\Users\chus6\Desktop\KKuTu-master\KKuTu-master\Server\lib\node_modules\pg\lib\connection.js:454:9) at Connection.parseMessage (C:\Users\chus6\Desktop\KKuTu-master\KKuTu-master\Server\lib\node_modules\pg\lib\connection.js:370:17) at Socket.&lt;anonymous> (C:\Users\chus6\Desktop\KKuTu-master\KKuTu-master\Server\lib\node_modules\pg\lib\connection.js:129:22) at Socket.emit (events.js:314:20) at addChunk (_stream_readable.js:297:12) at readableAddChunk (_stream_readable.js:272:9) at Socket.Readable.push (_stream_readable.js:213:10) at TCP.onStreamRead (internal/stream_base_commons.js:188:23)

라는 에러가 뜨며 그냥 서버랑 워커가 죽습니다..

connection.js 어디에 문제가 있는지 모르겠습니다.

lshqqytiger commented:
https://github.com/brianc/node-postgres/issues/1508

aslhanball commented:
그.. 아예 이젠 아무것도 안돼서 처음부터 다시하는중입니다... 너무 질문 많이한점 죄송합니다.

aslhanball commented:
[2024-01-20 10:15:02] ERROR OCCURRED ON THE MASTER!
:8080 [2024-1-20 10:15:02 ├F10: AM┤] ERROR: TypeError: Cannot read property 'url' of undefined TypeError: Cannot read property 'url' of undefined at WebSocketServer.&lt;anonymous> (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\Game\master.js:351:19) at WebSocketServer.emit (events.js:314:20) at C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:91:14 at completeHybiUpgrade2 (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:284:5) at completeHybiUpgrade1 (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:309:13) at WebSocketServer.handleHybiUpgrade (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:337:3) at WebSocketServer.handleUpgrade (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:173:26) at Server.WebSocketServer._onServerUpgrade (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:89:12) at Server.emit (events.js:314:20) at onParserExecuteCommon (_http_server.js:641:14)

서버는 작동하지만 에러가 뜨며 채널에 들어가지지 않습니다.

Preta-Crowz commented:
> [2024-01-20 10:15:02] ERROR OCCURRED ON THE MASTER! :8080 [2024-1-20 10:15:02 ├F10: AM┤] ERROR: TypeError: Cannot read property 'url' of undefined TypeError: Cannot read property 'url' of undefined at WebSocketServer.<anonymous> (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\Game\master.js:351:19) at WebSocketServer.emit (events.js:314:20) at C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:91:14 at completeHybiUpgrade2 (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:284:5) at completeHybiUpgrade1 (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:309:13) at WebSocketServer.handleHybiUpgrade (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:337:3) at WebSocketServer.handleUpgrade (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:173:26) at Server.WebSocketServer._onServerUpgrade (C:\Users\chus6\Desktop\KKuTu-master (2)\KKuTu-master\Server\lib\node_modules\ws\lib\WebSocketServer.js:89:12) at Server.emit (events.js:314:20) at onParserExecuteCommon (_http_server.js:641:14)
> 
> 서버는 작동하지만 에러가 뜨며 채널에 들어가지지 않습니다.

PR 병합이 힘들다면 PR 병합 대신 병합 원본 레포지토리를 포킹하는것이 권장됩니다.

aslhanball commented:
그럼 지금 저상태로는 해결하기 조금 복잡한건가요..

Preta-Crowz commented:
> 그럼 지금 저상태로는 해결하기 조금 복잡한건가요..

양쪽 코드가 제대로 병합이 안된것 같습니다. 어느 한쪽의 코드만 사용하는 경우에는 문제 없이 동작합니다.
직접 양쪽 코드 보면서 뭐가 문제인지 찾거냐, 그냥 레포지토리를 포킹해서 별도로 병합하는게 아닌 이상은 어렵습니다.
복잡하다기보다는, 귀찮은 작업입니다. (특히 파일을 다운로드만 해서 돌린다면, 몇만배는 더 귀찮습니다.)

aslhanball commented:
전 PR을 내려받아서 스크립트를 합치는 방법을 사용했습니다.

Preta-Crowz commented:
> 전 PR을 내려받아서 스크립트를 합치는 방법을 사용했습니다.

포킹 후 PR을 바로 진행하면 별도로 합치는 작업이 없습니다. 작업 방식 자체가 잘못되었을 가능성이 높습니다.

> 끄투 서버의 정상적인 운영을 위해서는 충분한 개발 실력을 요구합니다. 부족하다고 여겨질 경우, 다른 개발자를 구하거나 외주를 맡기시기 바랍니다. (#689)

aslhanball commented:
master.js:351:19
events.js:314:20
http_server.js:641:14

를 수정하면 해결될수도 있는건가요

d0ul commented:
> 전 PR을 내려받아서 스크립트를 합치는 방법을 사용했습니다.

소스코드를 내려받아서 합치는 행위는 적절하지 못합니다.
Git에서 Remote로 해당 레포지토리를 추가하신 후 git merge (리모트이름)/(브랜치이름)을 통해 합친 후 VS Code와 같은 Git 클라이언트에서 충돌을 해결하여 Grunt를 실행하셔야 합니다.

아울러 Git 사용 방법이나 끄투 코드에 대한 이해가 전무하신 것 같은데, 적절한 개발 실력을 키우고 오시길 부탁드립니다.
남들의 도움을 받아 구축된 끄투 프리서버는 결국 새로운 기능을 만들거나 서버에 오류가 발생할때마다 계속하여 남의 도움을 필요로 할 것이고, 그런 프리서버는 오래 가지 못합니다.

aslhanball commented:
그러면 어떻게 하면 되나요

Preta-Crowz commented:
> 그러면 어떻게 하면 되나요

이미 방법을 말씀드렸습니다.

d0ul commented:
> 그러면 어떻게 하면 되나요

제가 답변을 너무 어렵게 한 것 같네요.
한마디로 요약하자면 **코딩 실력이 부족하면 끄투 서버를 운영하지 마라**입니다.

또한 간단한 질문은 ChatGPT로 부탁드립니다
# 소스코드 병합 방법
새로운 원격(remote) 추가:

git remote add 다른포크 https://github.com/다른계정/다른레포.git
여기서 다른포크는 가져올 다른 레포지토리에 대한 별칭입니다. URL은 가져올 레포지토리의 주소로 바꿔야 합니다.
원격에서 변경 사항 가져오기:

git fetch 다른포크
로컬 브랜치에 가져온 변경 사항 병합:

git merge 다른포크/브랜치이름
여기서 브랜치이름은 가져온 레포지토리의 특정 브랜치 이름으로 바꿔야 합니다.
또는 rebase를 사용하여 변경 사항을 쌓을 수도 있습니다:

git rebase 다른포크/브랜치이름
이제 다른 레포지토리의 변경 사항이 로컬 레포지토리에 반영되었습니다. 이 변경 사항을 로컬 브랜치에서 작업하거나 다른 브랜치로 병합할 수 있습니다.

aslhanball commented:
> > 그럼 지금 저상태로는 해결하기 조금 복잡한건가요..
> 
> 양쪽 코드가 제대로 병합이 안된것 같습니다. 어느 한쪽의 코드만 사용하는 경우에는 문제 없이 동작합니다. 직접 양쪽 코드 보면서 뭐가 문제인지 찾거냐, 그냥 레포지토리를 포킹해서 별도로 병합하는게 아닌 이상은 어렵습니다. 복잡하다기보다는, 귀찮은 작업입니다. (특히 파일을 다운로드만 해서 돌린다면, 몇만배는 더 귀찮습니다.)

보니까 별거 아니였네요 너무 질문 많이해서 죄송합니다
